File: data/s11/scripts/check-cross-tier-solo.py
import json
import os
import re

root = os.path.join(os.path.dirname(__file__), "..")
fixtures = json.load(open(os.path.join(root, "fixtures.json"), encoding="utf-8"))
players = json.load(open(os.path.join(root, "players.json"), encoding="utf-8"))["players"]
tier_by = {(p.get("displayName") or "").strip(): (p.get("tier") or "").strip() for p in players}


def plan(r):
    rd = next((x for x in fixtures.get("rounds", []) if x.get("round") == r), None)
    return {s["slot"]: s for s in (rd or {}).get("slotPlan", [])}


mismatches = []
for f in sorted(os.listdir(os.path.join(root, "results"))):
    if not re.match(r"^s11-r\d+-m\d+\.json$", f):
        continue
    doc = json.load(open(os.path.join(root, "results", f), encoding="utf-8"))
    rnum = doc.get("round")
    pmap = plan(rnum)
    for mu in doc.get("matchups", []):
        for sl in mu.get("slots", []):
            sp = pmap.get(sl.get("slot"))
            if not sp or sp.get("format") != "1v1" or not sp.get("tierLine"):
                continue
            st = sp["tierLine"].strip()
            for side in ("teamA", "teamB"):
                for row in sl.get(side) or []:
                    name = (row.get("displayName") or row.get("playerName") or "").strip()
                    if not name:
                        continue
                    pt = tier_by.get(name, "")
                    if pt != st:
                        mismatches.append(
                            {
                                "file": f,
                                "fixtureId": mu.get("fixtureId"),
                                "round": rnum,
                                "slot": sl.get("slot"),
                                "slotTier": st,
                                "map": sp.get("map"),
                                "player": name,
                                "playerTier": pt or "(미등록)",
                            }
                        )

for m in mismatches:
    print(
        f"{m['round']}R #{m['slot']} {m['slotTier']}({m['map']}) | "
        f"{m['player']} 선수티어={m['playerTier']} | {m['fixtureId']}"
    )
print(f"--- 총 {len(mismatches)}건 ---")

File: data/s11/scripts/check-cross-tier-solo.mjs
import { readFileSync, readdirSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const root = join(dirname(fileURLToPath(import.meta.url)), '..');
const fixtures = JSON.parse(readFileSync(join(root, 'fixtures.json'), 'utf8'));
const { players } = JSON.parse(readFileSync(join(root, 'players.json'), 'utf8'));

const tierByName = new Map();
for (const p of players) {
  const n = (p.displayName || '').trim();
  if (n) tierByName.set(n, (p.tier || '').trim());
}

function slotPlanByRound(roundNum) {
  const rd = (fixtures.rounds || []).find((x) => x.round === roundNum);
  const m = new Map();
  for (const s of rd?.slotPlan || []) m.set(s.slot, s);
  return m;
}

function lineName(row) {
  if (!row) return '';
  return (row.displayName || row.playerName || '').trim();
}

const resultsDir = join(root, 'results');
const files = readdirSync(resultsDir)
  .filter((f) => /^s11-r\d+-m\d+\.json$/.test(f))
  .sort();

const mismatches = [];

for (const file of files) {
  const doc = JSON.parse(readFileSync(join(resultsDir, file), 'utf8'));
  const round = doc.round;
  if (round == null) continue;
  const plan = slotPlanByRound(round);

  for (const mu of doc.matchups || []) {
    for (const sl of mu.slots || []) {
      const sp = plan.get(sl.slot);
      if (!sp || sp.format !== '1v1' || !sp.tierLine) continue;

      const slotTier = sp.tierLine.trim();
      for (const side of ['teamA', 'teamB']) {
        for (const row of sl[side] || []) {
          const name = lineName(row);
          if (!name) continue;
          const playerTier = tierByName.get(name);
          if (!playerTier) {
            mismatches.push({
              file,
              fixtureId: mu.fixtureId,
              round,
              slot: sl.slot,
              slotTier,
              map: sp.map,
              player: name,
              playerTier: '(미등록)',
              kind: 'unknown',
            });
            continue;
          }
          if (playerTier !== slotTier) {
            mismatches.push({
              file,
              fixtureId: mu.fixtureId,
              round,
              slot: sl.slot,
              slotTier,
              map: sp.map,
              player: name,
              playerTier,
              kind: 'cross-tier',
            });
          }
        }
      }
    }
  }
}

console.log(JSON.stringify(mismatches, null, 2));
console.error(`\n총 ${mismatches.length}건 (cross-tier: ${mismatches.filter((x) => x.kind === 'cross-tier').length})`);
